plugins/inputs/modbus/request.go

```diff
@@ -53,16 +57,23 @@ func groupFieldsToRequests(fields []field, tags map[string]string, maxBatchSize
 			current.length += f.length
 			if !f.omit {
 				// Omit adding the field but use it for constructing the request.
-				current.fields = append(current.fields, f)
+				if isNonEmptyRequest(current) {
+					current.fields = append(current.fields, f)
+				} else {
+					current = newRequest(f, tags)
+				}
 			}
 			continue
 		}
 
 		// Finish the current request, add it to the list and construct a new one
-		requests = append(requests, current)
+		if isNonEmptyRequest(current) {
+			requests = append(requests, current)
+		}
 		current = newRequest(f, tags)
 	}
-	requests = append(requests, current)
-
+	if isNonEmptyRequest(current) {
+		requests = append(requests, current)
+	}
```

Comment:
Instead of all of this I would just check if the request will fill a field and delete it otherwise
```suggestion
	requests = append(requests, current)

	// Filter out completely empty requests
	nonEmptyRequests := make([]request, 0, len(requests))
	for _, request := range requests {
		if len(request.fields) != 0 {
			nonEmptyRequests = append(nonEmptyRequests, request)
		}
	}
```
and then return `nonEmptyRequests` below.